flink-connector-dynamodb/src/main/java/org/apache/flink/streaming/connectors/dynamodb/sink/DynamoDbSinkWriter.java

```diff
@@ -0,0 +1,306 @@
+                container.put(keyBuilder.build(req), req);
+            }
+            items.addAll(container.values());
+        }
+
+        CompletableFuture<BatchWriteItemResponse> future =
+                client.batchWriteItem(
+                        BatchWriteItemRequest.builder()
+                                .requestItems(ImmutableMap.of(tableName, items))
+                                .build());
+
+        future.whenComplete(
+                (response, err) -> {
+                    if (err != null) {
+                        handleFullyFailedRequest(err, requestEntries, requestResultConsumer);
+                    } else if (!CollectionUtil.isNullOrEmpty(response.unprocessedItems())) {
+                        handlePartiallyUnprocessedRequest(response, requestResultConsumer);
+                    } else {
+                        requestResultConsumer.accept(Collections.emptyList());
+                    }
+                });
+    }
+
+    private void handlePartiallyUnprocessedRequest(
+            BatchWriteItemResponse response, Consumer<List<DynamoDbWriteRequest>> requestResult) {
+        List<DynamoDbWriteRequest> unprocessed = new ArrayList<>();
+
+        for (String tableName : response.unprocessedItems().keySet()) {
+            for (WriteRequest writeRequest : response.unprocessedItems().get(tableName)) {
+                unprocessed.add(convertToDynamoDbWriteRequest(writeRequest));
+            }
+        }
+
+        LOG.warn("DynamoDB Sink failed to persist and will retry {} entries.", unprocessed.size());
+        numRecordsReturnedAsUnprocessedCounter.inc(unprocessed.size());
+
+        requestResult.accept(unprocessed);
+    }
+
+    private void handleFullyFailedRequest(
+            Throwable err,
+            List<DynamoDbWriteRequest> requestEntries,
+            Consumer<List<DynamoDbWriteRequest>> requestResult) {
+        LOG.warn(
+                "DynamoDB Sink failed to persist and will retry {} entries.",
+                requestEntries.size(),
+                err);
+        numRecordsSendErrorsCounter.inc(requestEntries.size());
+
+        if (isRetryable(err.getCause())) {
+            requestResult.accept(requestEntries);
+        }
+    }
+
+    private boolean isRetryable(Throwable err) {
+        if (!DYNAMODB_FATAL_EXCEPTION_CLASSIFIER.isFatal(err, getFatalExceptionCons())) {
+            return false;
+        }
+        if (failOnError) {
+            getFatalExceptionCons()
+                    .accept(new DynamoDbSinkException.DynamoDbSinkFailFastException(err));
+            return false;
+        }
+
+        return true;
+    }
+
+    @Override
+    protected long getSizeInBytes(DynamoDbWriteRequest requestEntry) {
+        // dynamodb calculates item size as a sum of all attributes and all values, but doing so on
+        // every operation may be too expensive, so this is just an estimate
+        return requestEntry.getItem().toString().getBytes(StandardCharsets.UTF_8).length;
```

Comment:
As discussed offline, it is not practical to compute the `sizeInBytes` given that:
- We do not know how the client serializes the request payload
- This will be an expensive operation and we would potentially be serializing each record twice

We agree to:
- Return a dummy constant value here
- Follow up with an improvement to make this flush strategy and validation optional

The failure modes are already covered:
- Total batch size too large: this will be handled organically by AIMD
- Single `DynamoDbWriteRequest` too large: if we try to send a single request that is too large it will result in a fatal validation exception

Reply:
This was fixed. I set to 0 and updated builder class to raise exceptions if corresponding parameters were used